processing/src/main/java/io/druid/query/QueryMetrics.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query;
+
+import com.metamx.emitter.service.ServiceEmitter;
+
+import java.util.Map;
+
+/**
+ * Abstraction wrapping {@link com.metamx.emitter.service.ServiceMetricEvent.Builder} and allowing to control what
+ * metrics are actually emitted, what dimensions do they have, etc.
+ *
+ *
+ * Goals of QueryMetrics
```

Comment:
When reading these docs, I wondered, _who_ is supposed to have the kinds of control you mention?

- The Druid project itself -- is this a tool to help us write code more easily?
- The operator of a Druid cluster -- is this a tool to help Druid cluster operators control metrics emitted by built-in query types? If so, how is this done: through config, through writing a site-specific extension, or through forking Druid?
- Authors of extensions involving new Druid query types -- is this a tool to help implement metrics emission for new queries?